gix-worktree/src/index/status/worktree.rs

```diff
@@ -0,0 +1,272 @@
+use std::io;
+use std::marker::PhantomData;
+use std::path::Path;
+
+use crate::index::status::diff::{self, Diff};
+use crate::index::status::{Collector, Status};
+use crate::{fs, read};
+use bstr::BStr;
+use filetime::FileTime;
+use gix_features::parallel::{in_parallel_chunks_if, Reduce};
+use gix_index as index;
+use gix_path as path;
+
+mod untracked;
+
+/// The error returned by [`compare_to_index()`].
+#[derive(Debug, thiserror::Error)]
+#[allow(missing_docs)]
+pub enum Error {
+    #[error("Could not convert path to UTF8")]
+    IllformedUtf8,
+    #[error("The clock was off when reading file related metadata after updating a file on disk")]
+    Time(#[from] std::time::SystemTimeError),
+    #[error("IO error while writing blob or reading file metadata or changing filetype")]
+    Io(#[from] io::Error),
+}
+
+#[derive(Clone)]
+/// Options that control how the index status of a worktree is computed
+pub struct Options {
+    /// capabilities of the file system
+    pub fs: crate::fs::Capabilities,
+    /// If set, don't use more than this amount of threads.
+    /// Otherwise, usually use as many threads as there are logical cores.
+    /// A value of 0 is interpreted as no-limit
+    pub thread_limit: Option<usize>,
+    /// If true, default false, try don't abort on first error which isn't
+    /// due to a conflict.
+    /// The checkout operation will never fail, but count the encountered errors instead along with their paths.
+    pub keep_going: bool,
+    /// Options that control how stat comparisons are made
+    /// when checking if a file is fresh
+    pub stat: index::entry::stat::Options,
+    // /// A group of attribute patterns that are applied globally, i.e. aren't rooted within the repository itself.
+    // pub attribute_globals: gix_attributes::MatchGroup<Attributes>,
+    /// Untracked files that were added to the index with git add but not yet committed
+    /// are marked with special flags and usually receive special treatment. If this option
+    /// is enabled (default true) added events are generated for these files, otherwise
+    /// these files are treated the same as other entries
+    pub check_added: bool,
+}
+
+impl Default for Options {
+    fn default() -> Self {
+        Options {
+            fs: fs::Capabilities::default(),
+            thread_limit: None,
+            keep_going: false,
+            stat: index::entry::stat::Options::default(),
+            check_added: true,
+        }
+    }
+}
+
+/// Calculates the status of worktree
```

Comment:
yeah I agree I renamed it to `index::status::worktree:changes_to_obtain` for now. Feel free to experiment with what you feel works best. I would be in favor of moving this to `gix-index::status` or `gix-index::diff` then we could have `gix-index::status::worktree::changes_to_obtain` or `gix-index::status::changes_to_obtain_worktreee` (the latter is better but precludes us from making `gix-index::status::worktree` public which would be unfortunate since there is an error type etc in there. I will let you figure that out :D